cmd/cmdoptions/client.go

```diff
@@ -1,14 +1,29 @@
 package cmdoptions
 
 import (
+	"context"
 	"crypto/tls"
 	"errors"
 	"fmt"
+	"os"
+
 	"github.com/spf13/pflag"
 	"go.temporal.io/sdk/client"
 	"go.uber.org/zap"
 )
 
+type authTokenProvider struct {
+	token string
+}
+
+func (p *authTokenProvider) GetHeaders(ctx context.Context) (map[string]string, error) {
+	return map[string]string{"Authorization": p.token}, nil
```

Comment:
You might want to move the getenv call to be in here so, if necessary, the value can be changed without a restart